politeiawww/backend.go

```diff
@@ -2123,13 +2130,13 @@ func (b *backend) ProcessComment(c www.NewComment, user *database.User) (*www.Ne
 	if err != nil {
 		return nil, err
 	}
-	ncrWWW := b.convertDecredNewCommentReplyToWWWNewCommentReply(*ncr)
 
-	// Add comment to cache
-	b.Lock()
-	defer b.Unlock()
+	ncrWWW := b.convertDecredNewCommentReplyToWWWNewCommentReply(*ncr)
 
-	b.inventory[ncrWWW.Comment.Token].comments[ncrWWW.Comment.CommentID] = ncrWWW.Comment
+	err = b._setRecordComment(ncrWWW.Comment)
+	if err != nil {
+		return nil, fmt.Errorf("setRecordComment %v", err)
```

Comment:
No unlock causing a deadlock. Why does the function fail? I feel like it should based on the prior code.